modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/action/DLConfigurationAction.java

```diff
@@ -55,6 +68,42 @@ public void include(
 		super.include(portletConfig, httpServletRequest, httpServletResponse);
 	}
 
+	@Override
+	protected void postProcess(
+			long companyId, PortletRequest portletRequest,
+			PortletPreferences portletPreferences)
+		throws PortalException {
+
+		super.postProcess(companyId, portletRequest, portletPreferences);
+
+		try {
+			String[] displayViews = StringUtil.split(
+				portletPreferences.getValue("displayViews", null));
+
+			portletPreferences.setValues("displayViews", displayViews);
+		}
+		catch (ReadOnlyException readOnlyException) {
+			throw new SystemException(readOnlyException);
+		}
+
+		long rootFolderId = GetterUtil.getLong(
+			portletPreferences.getValue("rootFolderId", null));
+
+		long selectedRepositoryId = GetterUtil.getLong(
+			portletPreferences.getValue("selectedRepositoryId", null));
+
+		try {
+			_setPortletPreferences(
+				portletPreferences, rootFolderId, selectedRepositoryId);
+
+			portletPreferences.reset("rootFolderId");
+			portletPreferences.reset("selectedRepositoryId");
```

Comment:
Why not move these lines into the `_setPortletPreferences` method?

Reply:
See change here: [dc1c987](https://github.com/liferay-content-management/liferay-portal/pull/6244/commits/dc1c987eebb0d916abcb1cb9c56cff5f29648b86)